core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala

```diff
@@ -81,8 +81,9 @@ private[spark] class TaskResultGetter(sparkEnv: SparkEnv, scheduler: TaskSchedul
                   taskSetManager, tid, TaskState.FINISHED, TaskResultLost)
                 return
               }
+              // TODO(josh): assumption that there is only one chunk here is a hack
               val deserializedResult = serializer.get().deserialize[DirectTaskResult[_]](
-                serializedTaskResult.get)
+                serializedTaskResult.get.getChunks().head)
```

Comment:
Since there's only ever (in expectation) going to be one chunk here, calling `.toByteBuffer` should have no real performance or correctness issues here, so I'll just do that instead.